link.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from scrapy.linkextractors import LinkExtractor
+from scrapy.http import HtmlResponse, TextResponse, Response
+from six.moves.urllib.parse import urlparse
+import glob
+from timeit import default_timer as timer
+import click
+import io
+
+
+def main():
+    start = timer()
+
+    url = 'http://scrapinghub.com/'
+    link_extractor = LinkExtractor()
+    total = 0
+    for files in glob.glob('sites/index*'):
+
+        f = (io.open(files, "r", encoding="ISO-8859-1"))
```

Comment:
I see - that means that files themselves are in latin-1 encoding. Could you also include some description about this files (maybe in the README) - were are the urls form, and how did you download them? Are you sure they all use latin-1 encoding? Is re-downloading them in utf-8 (e.g. with some scrapy spider) an option?